File: api/Dashboard.API/Models/Service/ServiceModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Dashboard.API.Models.ServiceUser;
using Dashboard.API.Models.Widget;

namespace Dashboard.API.Models.Service
{
    public sealed class ServiceModel
    {
        #region CONSTRUCTORS

        public ServiceModel()
        {
            ServiceId = Guid.NewGuid();
        }

        #endregion CONSTRUCTORS
        
        #region FIELDS
        public Guid ServiceId { get; set; }

        public string ServiceName { get; set; }
        
        public string UrlImage { get; set; }
        
        public string AuthorizeUrl { get; set; }
        
        public string AccessToken { get; set; }

        public List<WidgetModel> Widgets { get; set; }

        public List<ServiceUserModel> ServiceUsers { get; set; }
        
        #endregion FIELDS
    }
}

File: Dashboard/Dashboard/Sources/Models/Service/ServiceModel.swift
//
//  microServiceModel.swift
//  Dashboard
//
//  Created by Valentin Mille on 21/11/2020.
//

import Foundation

struct ServiceModel: Decodable {
    let serviceId: String
    let serviceName: String
    let urlImage: String
    let authorizeUrl: String?
    let accessToken: String?
    var oauthToken: String?
    var refreshToken: String?
}
